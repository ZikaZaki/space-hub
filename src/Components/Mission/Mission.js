import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import Badge from 'react-bootstrap/Badge';
import { toggleMission } from '../../redux/missions/missions';
// import styles
import styles from './Mission.module.css';

const Mission = ({
  id,
  name,
  description,
  joined,
}) => {
  const dispatch = useDispatch();

  return (
    <>
      <th scope="row"><h5>{ name }</h5></th>
      <td>
        <p>{ description }</p>
      </td>
      <td>
        { joined ? <Badge className="m-2" bg="active" style={{ backgroundColor: '#18a2b8' }}>Active Member</Badge>
          : <Badge className="m-2 " bg="secondary">NOT A MEMBER</Badge> }
        {' '}
      </td>
      <td>
        { joined
          ? (
            <button
              type="button"
              className={styles['leave-btn']}
              onClick={() => dispatch(toggleMission(id))}
            >
              Leave Mission
            </button>
          )
          : (
            <button
              type="button"
              className={styles['join-btn']}
              onClick={() => dispatch(toggleMission(id))}
            >
              Join Mission
            </button>
          )}
        {' '}
      </td>
    </>
  );
};

Mission.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  joined: PropTypes.bool.isRequired,
};

export default Mission;
