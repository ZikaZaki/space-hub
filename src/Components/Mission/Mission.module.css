.join-btn {
  text-align: center;
  color: rgb(113, 109, 109);
  background-color: transparent;
  border-radius: 8px;
  border: 1px solid rgb(113, 109, 109);
  outline: none;
  min-width: 10rem;
  min-height: 2.5rem;
  padding: 0;
  margin: 1rem;
}

.join-btn:hover {
  color: #fff;
  background-color: rgb(113, 109, 109);
}

.leave-btn {
  text-align: center;
  color: red;
  background-color: transparent;
  border-radius: 8px;
  border: 1px solid red;
  outline: none;
  min-width: 10rem;
  min-height: 2.5rem;
  padding: 0;
  margin: 1rem;
}

.leave-btn:hover {
  color: #fff;
  background-color: red;
}
