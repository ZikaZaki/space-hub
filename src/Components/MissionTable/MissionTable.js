import React from 'react';
import { useSelector } from 'react-redux';
import Table from 'react-bootstrap/Table';
import Mission from '../Mission/Mission';
// import styles
import styles from './MissionTable.module.css';

const MissionTable = () => {
  // Get missions from the store
  const missions = useSelector((state) => state.missions);

  return (
    <div className={styles['missions-container']} data-testid="missions-container">
      <Table striped bordered hover responsive="xl" size="sm">
        <thead>
          <tr>
            <th className="p-2" scope="col"><h4>Mission</h4></th>
            <th className="p-2" scope="col"><h4>Description</h4></th>
            <th className="p-2" scope="col"><h4>Status</h4></th>
            <th className="p-2" scope="col"> </th>
          </tr>
        </thead>
        <tbody>
          { missions && missions.map((mission) => (
            <tr key={mission.id}>
              <Mission
                id={mission.id}
                name={mission.name}
                description={mission.description}
                joined={mission.joined}
              />
            </tr>
          ))}
        </tbody>
      </Table>
    </div>
  );
};

export default MissionTable;
