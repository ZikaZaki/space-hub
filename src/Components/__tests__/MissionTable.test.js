import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Mission from '../Mission/Mission';
import MissionTable from '../MissionTable/MissionTable';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if the component renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MissionTable>
        <Mission
          id="9D1B7E0"
          name="FalconSat"
          description="The FalconSat is an American nanosatellite built by SpaceX"
          joined={false}
        />
      </MissionTable>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('the component should render a div container', () => {
  render(
    <Provider store={store}>
      <MissionTable>
        <Mission
          id="9D1B7E0"
          name="FalconSat"
          description="The FalconSat is an American nanosatellite built by SpaceX"
          joined={false}
        />
      </MissionTable>
    </Provider>,
  );

  const mission = screen.getByTestId('missions-container');

  expect(mission).toBeInTheDocument();
});
