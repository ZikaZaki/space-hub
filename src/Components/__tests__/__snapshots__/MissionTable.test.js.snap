// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component renders correctly 1`] = `
<div
  className="missions-container"
  data-testid="missions-container"
>
  <div
    className="table-responsive-xl"
  >
    <table
      className="table table-sm table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th
            className="p-2"
            scope="col"
          >
            <h4>
              Mission
            </h4>
          </th>
          <th
            className="p-2"
            scope="col"
          >
            <h4>
              Description
            </h4>
          </th>
          <th
            className="p-2"
            scope="col"
          >
            <h4>
              Status
            </h4>
          </th>
          <th
            className="p-2"
            scope="col"
          >
             
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
